omnigibson/utils/bddl_utils.py

```diff
@@ -89,9 +127,810 @@ def process_single_condition(condition):
     "open": get_unary_predicate_for_state(object_states.Open, "open"),
     "toggled_on": get_unary_predicate_for_state(object_states.ToggledOn, "toggled_on"),
     "frozen": get_unary_predicate_for_state(object_states.Frozen, "frozen"),
+    "future": ObjectStateFuturePredicate,
+    "real": ObjectStateRealPredicate,
 }
 
+KINEMATIC_STATES_BDDL = frozenset([state.__name__.lower() for state in _KINEMATIC_STATE_SET])
+
+# Load substance and object mapping
+with open(f"{bddl.__path__[0]}/../substance_synset_mapping.json", "r") as f:
+    SUBSTANCE_SYNSET_MAPPING = json.load(f)
+
+
+# BEHAVIOR-related
+OBJECT_TAXONOMY = ObjectTaxonomy() #hierarchy_type="b1k")
```

Comment:
`SUBSTANCE_SYNSET_MAPPING` should be subsumed into `OBJECT_TAXONOMY`

Reply:
put TODO

Reply:
done